File: packages/sdk/dotnet/Modules/AgentsApi.cs
using System.Text.Json;

namespace Ihui.AI;

/// <summary>
/// Agent 模块 — 列表 / 调用 / 高级执行 / Pipeline / 并行 / 任务分解。
/// </summary>
/// <remarks>
/// 端点(12 个):
/// <list type="bullet">
///   <item>GET  /v1/agents</item>
///   <item>GET  /v1/agents/:id</item>
///   <item>POST /v1/agents/:id/call</item>
///   <item>POST /v1/agents/execute(高级执行)</item>
///   <item>POST /v1/agents/execute/stream(SSE 流式执行)</item>
///   <item>GET  /v1/agents/tasks/:id/status</item>
///   <item>POST /v1/agents/tasks/:id/cancel</item>
///   <item>GET  /v1/agents/sessions</item>
///   <item>DELETE /v1/agents/sessions/:id</item>
///   <item>POST /v1/agents/pipeline</item>
///   <item>POST /v1/agents/parallel</item>
///   <item>POST /v1/agents/decompose</item>
/// </list>
/// </remarks>
public sealed class AgentsApi
{
    private readonly BaseClient _client;

    internal AgentsApi(BaseClient client) => _client = client;

    /// <summary>GET /v1/agents(Agent 列表)。</summary>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>Agent 列表(JsonElement)</returns>
    public Task<JsonElement?> ListAsync(CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("GET", "/agents", null, cancellationToken);

    /// <summary>GET /v1/agents/:id(Agent 详情)。</summary>
    /// <param name="id">Agent ID</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>Agent 详情(JsonElement)</returns>
    public Task<JsonElement?> GetAsync(string id, CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("GET", "/agents/" + BaseClient.Encode(id), null, cancellationToken);

    /// <summary>POST /v1/agents/:id/call(调用 Agent)。</summary>
    /// <param name="id">Agent ID</param>
    /// <param name="req">请求体(input / context 等)</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>调用结果(JsonElement)</returns>
    public Task<JsonElement?> CallAsync(string id, object req, CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("POST", "/agents/" + BaseClient.Encode(id) + "/call", req, cancellationToken);

    /// <summary>POST /v1/agents/execute(高级执行,支持 PermissionGuard)。</summary>
    /// <param name="req">Agent 执行请求</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>Agent 执行响应</returns>
    public Task<AgentExecuteResponse?> ExecuteAsync(AgentExecuteRequest req, CancellationToken cancellationToken = default)
        => _client.RequestAsync<AgentExecuteResponse>("POST", "/agents/execute", req, cancellationToken);

    /// <summary>POST /v1/agents/execute/stream(SSE 流式执行)。</summary>
    /// <param name="req">Agent 执行请求</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>IAsyncEnumerable,逐事件 yield JsonElement</returns>
    public IAsyncEnumerable<JsonElement> ExecuteStreamAsync(AgentExecuteRequest req, CancellationToken cancellationToken = default)
        => _client.RequestStreamAsync("POST", "/agents/execute/stream", req, cancellationToken);

    /// <summary>GET /v1/agents/tasks/:id/status(任务状态)。</summary>
    /// <param name="taskId">任务 ID</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>任务状态(JsonElement)</returns>
    public Task<JsonElement?> GetTaskStatusAsync(string taskId, CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("GET", "/agents/tasks/" + BaseClient.Encode(taskId) + "/status", null, cancellationToken);

    /// <summary>POST /v1/agents/tasks/:id/cancel(取消任务)。</summary>
    /// <param name="taskId">任务 ID</param>
    /// <param name="cancellationToken">取消令牌</param>
    public Task CancelTaskAsync(string taskId, CancellationToken cancellationToken = default)
        => _client.RequestRawAsync("POST", "/agents/tasks/" + BaseClient.Encode(taskId) + "/cancel", null, cancellationToken)!;

    /// <summary>GET /v1/agents/sessions(会话列表)。</summary>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>会话列表(JsonElement)</returns>
    public Task<JsonElement?> ListSessionsAsync(CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("GET", "/agents/sessions", null, cancellationToken);

    /// <summary>DELETE /v1/agents/sessions/:id(删除会话)。</summary>
    /// <param name="id">会话 ID</param>
    /// <param name="cancellationToken">取消令牌</param>
    public Task DeleteSessionAsync(string id, CancellationToken cancellationToken = default)
        => _client.RequestRawAsync("DELETE", "/agents/sessions/" + BaseClient.Encode(id), null, cancellationToken)!;

    /// <summary>POST /v1/agents/pipeline(Pipeline 编排)。</summary>
    /// <param name="req">请求体</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>Pipeline 执行结果(JsonElement)</returns>
    public Task<JsonElement?> PipelineAsync(object req, CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("POST", "/agents/pipeline", req, cancellationToken);

    /// <summary>POST /v1/agents/parallel(并行执行)。</summary>
    /// <param name="req">请求体</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>并行执行结果(JsonElement)</returns>
    public Task<JsonElement?> ParallelAsync(object req, CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("POST", "/agents/parallel", req, cancellationToken);

    /// <summary>POST /v1/agents/decompose(任务分解)。</summary>
    /// <param name="req">Agent 执行请求</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>任务分解结果(JsonElement,含 taskId / subtasks)</returns>
    public Task<JsonElement?> DecomposeAsync(AgentExecuteRequest req, CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("POST", "/agents/decompose", req, cancellationToken);
}

File: packages/sdk/java/src/main/java/com/ihui/ai/sdk/module/AgentsApi.java
package com.ihui.ai.sdk.module;

import com.fasterxml.jackson.databind.JsonNode;
import com.ihui.ai.sdk.BaseClient;
import com.ihui.ai.sdk.StreamResponse;
import com.ihui.ai.sdk.model.AgentExecuteRequest;
import com.ihui.ai.sdk.model.AgentExecuteResponse;

import java.util.Map;

/**
 * Agent 模块 — 列表 / 调用 / 高级执行 / Pipeline / 并行 / 任务分解。
 *
 * <p>端点(12 个):
 * <ul>
 *   <li>GET  /v1/agents</li>
 *   <li>GET  /v1/agents/:id</li>
 *   <li>POST /v1/agents/:id/call</li>
 *   <li>POST /v1/agents/execute(高级执行)</li>
 *   <li>POST /v1/agents/execute/stream(SSE 流式执行)</li>
 *   <li>GET  /v1/agents/tasks/:id/status</li>
 *   <li>POST /v1/agents/tasks/:id/cancel</li>
 *   <li>GET  /v1/agents/sessions</li>
 *   <li>DELETE /v1/agents/sessions/:id</li>
 *   <li>POST /v1/agents/pipeline</li>
 *   <li>POST /v1/agents/parallel</li>
 *   <li>POST /v1/agents/decompose</li>
 * </ul>
 */
public final class AgentsApi {

    private final BaseClient client;

    /**
     * 构造 AgentsApi。
     *
     * @param client 底层 BaseClient
     */
    public AgentsApi(BaseClient client) {
        this.client = client;
    }

    /**
     * GET /v1/agents(Agent 列表)。
     *
     * @return Agent 列表(JsonNode)
     */
    public JsonNode list() {
        return client.request("GET", "/agents", null, JsonNode.class);
    }

    /**
     * GET /v1/agents/:id(Agent 详情)。
     *
     * @param id Agent ID
     * @return Agent 详情(JsonNode)
     */
    public JsonNode get(String id) {
        return client.request("GET", "/agents/" + BaseClient.encode(id), null, JsonNode.class);
    }

    /**
     * POST /v1/agents/:id/call(调用 Agent)。
     *
     * @param id  Agent ID
     * @param req 请求体(input / context 等)
     * @return 调用结果(JsonNode)
     */
    public JsonNode call(String id, Map<String, Object> req) {
        return client.request("POST", "/agents/" + BaseClient.encode(id) + "/call",
                req, JsonNode.class);
    }

    /**
     * POST /v1/agents/execute(高级执行,支持 PermissionGuard)。
     *
     * @param req Agent 执行请求
     * @return Agent 执行响应
     */
    public AgentExecuteResponse execute(AgentExecuteRequest req) {
        return client.request("POST", "/agents/execute", req, AgentExecuteResponse.class);
    }

    /**
     * POST /v1/agents/execute/stream(SSE 流式执行)。
     *
     * <p>用法:
     * <pre>
     * try (StreamResponse stream = client.agents.executeStream(req)) {
     *     while (stream.hasNext()) {
     *         JsonNode event = stream.next();
     *         // 处理事件
     *     }
     * }
     * </pre>
     *
     * @param req Agent 执行请求
     * @return StreamResponse(必须用 try-with-resources 关闭)
     */
    public StreamResponse executeStream(AgentExecuteRequest req) {
        return client.requestStream("POST", "/agents/execute/stream", req);
    }

    /**
     * GET /v1/agents/tasks/:id/status(任务状态)。
     *
     * @param taskId 任务 ID
     * @return 任务状态(JsonNode)
     */
    public JsonNode getTaskStatus(String taskId) {
        return client.request("GET", "/agents/tasks/" + BaseClient.encode(taskId) + "/status",
                null, JsonNode.class);
    }

    /**
     * POST /v1/agents/tasks/:id/cancel(取消任务)。
     *
     * @param taskId 任务 ID
     */
    public void cancelTask(String taskId) {
        client.requestRaw("POST", "/agents/tasks/" + BaseClient.encode(taskId) + "/cancel", null);
    }

    /**
     * GET /v1/agents/sessions(会话列表)。
     *
     * @return 会话列表(JsonNode)
     */
    public JsonNode listSessions() {
        return client.request("GET", "/agents/sessions", null, JsonNode.class);
    }

    /**
     * DELETE /v1/agents/sessions/:id(删除会话)。
     *
     * @param id 会话 ID
     */
    public void deleteSession(String id) {
        client.requestRaw("DELETE", "/agents/sessions/" + BaseClient.encode(id), null);
    }

    /**
     * POST /v1/agents/pipeline(Pipeline 编排)。
     *
     * @param req 请求体
     * @return Pipeline 执行结果(JsonNode)
     */
    public JsonNode pipeline(Map<String, Object> req) {
        return client.request("POST", "/agents/pipeline", req, JsonNode.class);
    }

    /**
     * POST /v1/agents/parallel(并行执行)。
     *
     * @param req 请求体
     * @return 并行执行结果(JsonNode)
     */
    public JsonNode parallel(Map<String, Object> req) {
        return client.request("POST", "/agents/parallel", req, JsonNode.class);
    }

    /**
     * POST /v1/agents/decompose(任务分解)。
     *
     * @param req Agent 执行请求
     * @return 任务分解结果(JsonNode,含 taskId / subtasks)
     */
    public JsonNode decompose(AgentExecuteRequest req) {
        return client.request("POST", "/agents/decompose", req, JsonNode.class);
    }
}
